### ISSUE TITLE
App crashes when inserting emoji in TextField and cursor is behind emoji

### ISSUE THREAD

**Issue by nohli** [flutter/flutter#36267]
I just have the beta on my iPhone, so there are no flutter logs.

Another problem with textfields on iOS 13 beta: The word suggestions above the keyboard are gone - although the textfield has autocorrect enabled.

I wouldn't have said anything, because it might just be a problem of the iOS beta, but the app Things used to freeze until an app update and the app Brilliant crashed until an app update.

I'm using the latest flutter stable (same as beta) and the latest iOS 13 developer beta.

## Steps to Reproduce

1. Build an app with a textfield
2. Install on iOS 13 beta device
3. Put an emoji into the textfield and put the cursor behind the emoji

**Comment by pkkhandsome** [comment]
yes.  
It crashed when i inserted an Emoji and then switch to another keyboard mode.
flutter : 1.7.8 hotfix3 
ios13 beta3 / iphonex

**Comment by nohli** [comment]
same on iOS developer beta 4

if the cursor is before (left of) the emoji, everything is fine, if it's after, app crashes immediately

**Comment by justinstigall** [comment]
I have the exact same problem with my app. If your cursor is to the left of an Emoji in a text field it's fine, as soon as you move to the right or select all the app crashes.  My debug release on iOS 12.4 works fine.

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.7.8+hotfix.4, on Mac OS X 10.14.6 18G87, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
[✓] iOS tools - develop for iOS devices
[✓] Android Studio (version 3.4)
[✓] VS Code (version 1.37.1)
[✓] Connected device (1 available)

• No issues found!


Incident Identifier: 378F1F76-FB97-4CFF-BF9C-90044D5CF03A
Beta Identifier:     59C1703B-35A9-416E-9BF0-CBFB42597732
Hardware Model:      iPhone11,6
Process:             Runner [21742]
Path:                /private/var/containers/Bundle/Application/315D1233-139E-48CB-AEFA-0F0A06396B0B/Runner.app/Runner
Identifier:          com.bauxite.calorly
Version:             6 (1.1)
AppStoreTools:       10G3
Beta:                YES
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           com.bauxite.calorly [4906]


Date/Time:           2019-08-17 21:02:15.9679 -0500
Launch Time:         2019-08-17 21:02:13.6475 -0500
OS Version:          iPhone OS 13.0 (17A5556d)
Release Type:        Beta
Baseband Version:    2.00.01
Report Version:      104

Exception Type:  EXC_BAD_ACCESS (SIGSEGV)
Exception Subtype: KERN_INVALID_ADDRESS at 0x000000027ffffff4
VM Region Info: 0x27ffffff4 is not in any region.  Bytes after previous region: 2871312373  Bytes before following region: 12
      REGION TYPE                      START - END             [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      unused shlib __TEXT    00000001d0000000-00000001d4db4000 [ 77.7M] r--/r-- SM=COW  ... this process
--->  GAP OF 0xab24c000 BYTES
      MALLOC_NANO            0000000280000000-00000002a0000000 [512.0M] rw-/rwx SM=PRV  

Termination Signal: Segmentation fault: 11
Termination Reason: Namespace SIGNAL, Code 0xb
Terminating Process: exc handler [21742]
Triggered by Thread:  0

Thread 0 name:
Thread 0 Crashed:
0   libsystem_platform.dylib      	0x0000000183105694 _platform_memmove + 420
1   CoreFoundation                	0x000000018339918c __CFStringCreateImmutableFunnel3 + 2508 (CFString.c:1799)
2   CoreFoundation                	0x000000018339d310 CFStringCreateWithSubstring + 504 (CFString.c:0)
3   CoreFoundation                	0x00000001832db2f4 -[__NSCFString substringWithRange:] + 180 (CFObject.m:283)
4   UIKitCore                     	0x000000018711f0d8 -[TIDocumentState(UITextInputAdditions) _contextAfterPosition:inDocument:toBoundary:] + 332 (TIDocumentState+UIKBExtras.m:135)
5   UIKitCore                     	0x000000018711f254 -[TIDocumentState(UITextInputAdditions) initWithDocument:contextBoundary:] + 272 (TIDocumentState+UIKBExtras.m:150)
6   UIKitCore                     	0x000000018711e710 +[TIDocumentState(UITextInputAdditions) documentStateOfDocument:] + 60 (TIDocumentState+UIKBExtras.m:23)
7   UIKitCore                     	0x0000000187280188 -[UIKeyboardImpl syncDocumentStateToInputDelegateWithExecutionContext:] + 1152 (UIKeyboardImpl.m:7359)
8   UIKitCore                     	0x00000001872808c8 -[UIKeyboardImpl updateForChangedSelectionWithExecutionContext:] + 556 (UIKeyboardImpl.m:7484)
9   UIKitCore                     	0x00000001872b2b34 -[UIKeyboardTaskEntry execute:] + 188 (UIKeyboardTaskQueue.m:737)
10  UIKitCore                     	0x00000001872b1498 -[UIKeyboardTaskQueue continueExecutionOnMainThread] + 324 (UIKeyboardTaskQueue.m:305)
11  UIKitCore                     	0x0000000187275408 -[UIKeyboardImpl setDelegate:force:] + 4600 (UIKeyboardImpl.m:3972)
12  UIKitCore                     	0x0000000186f711e0 -[UIInputResponderController _reloadInputViewsForKeyWindowSceneResponder:] + 2164 (UIInputResponderController.m:692)
13  UIKitCore                     	0x0000000186f7094c -[UIInputResponderController _reloadInputViewsForResponder:] + 168 (UIInputResponderController.m:0)
14  UIKitCore                     	0x00000001874918ec -[UIResponder(UIResponderInputViewAdditions) reloadInputViews] + 132 (UIResponder.m:1537)
15  UIKitCore                     	0x000000018748c9fc -[UIResponder becomeFirstResponder] + 800 (UIResponder.m:328)
16  UIKitCore                     	0x00000001878e9788 -[UIView(Hierarchy) becomeFirstResponder] + 164 (UIView.m:11362)
17  Flutter                       	0x00000001032fa2e0 0x1032e0000 + 107232
18  Flutter                       	0x0000000103365a18 0x1032e0000 + 547352
19  Flutter                       	0x00000001033097e8 0x1032e0000 + 169960
20  Flutter                       	0x0000000103358940 0x1032e0000 + 493888
21  Flutter                       	0x0000000103314434 0x1032e0000 + 214068
22  Flutter                       	0x0000000103317bc8 0x1032e0000 + 228296
23  CoreFoundation                	0x0000000183378d78 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 32 (CFRunLoop.c:1766)
24  CoreFoundation                	0x0000000183378ab0 __CFRunLoopDoTimer + 928 (CFRunLoop.c:2357)
25  CoreFoundation                	0x0000000183378124 __CFRunLoopDoTimers + 280 (CFRunLoop.c:2512)
26  CoreFoundation                	0x0000000183372fcc __CFRunLoopRun + 1896 (CFRunLoop.c:0)
27  CoreFoundation                	0x000000018337253c CFRunLoopRunSpecific + 480 (CFRunLoop.c:3192)
28  GraphicsServices              	0x000000018d970534 GSEventRunModal + 108 (GSEvent.c:2246)
29  UIKitCore                     	0x000000018745eeb0 UIApplicationMain + 1940 (UIApplication.m:4687)
30  Runner                        	0x0000000102a999a4 main + 88 (main.m:7)
31  libdyld.dylib                 	0x00000001831f2f04 start + 4

**Comment by nohli** [comment]
**This is a serious issue since the app crashes immediately when entering emojis into a textfield.**

Furthermore, if the input of the textfield is immediately saved, you can't even restart the app because the screen where the textfield (including the emoji) appears will immediately crash again (by default the cursor appears behind the text, so it will always crash again).

In the changelog of the latest stable release it says:

> Large number of iOS features and fixes, including improved text editing and localization

https://github.com/flutter/flutter/wiki/Release-Notes-Flutter-1.7.8

I guess, either there is a new, undiscovered bug, or iOS 13 has some breaking change.

In the meantime, it would be nice to have a workaround, something like checking if the input is valid, otherwise not saving it - to not let the app crash.

**Comment by nohli** [comment]
Today I tried to make the textfields at least stable, so the apps couldn't crash any more...

I found out, that even if insert
```
inputFormatters: [
                WhitelistingTextInputFormatter(RegExp("[a-zA-Z]"))
              ],
```

it would still crash...

It not only crashed because of emojis, but also @ and some other symbols.

When the textfield already contains text, you can let the app crash with an @ although it is forbidden through WhitelistingTextInputFormatter.

**Comment by nohli** [comment]
Today, after eight iOS 13 (lucky number) developer betas, Apple already released a developer beta of iOS 13.1...and all the crashes are fixed (I've tested after every update).

**Comment by jingjiewei** [comment]
> Today, after eight iOS 13 (lucky number) developer betas, Apple already released a developer beta of iOS 13.1...and all the crashes are fixed (I've tested after every update).

Verified on iOS v13.1 and issue gone
